packages/calypso-build/webpack/minify.js

```diff
@@ -11,4 +34,16 @@ const TerserPlugin = require( 'terser-webpack-plugin' );
  * @param {Object} options Options passed to the terser plugin
  * @returns {Object[]}     Terser plugin object to be used in Webpack minification.
  */
-module.exports = options => [ new TerserPlugin( options ) ];
+module.exports = options => {
+	let terserOptions = options.terserOptions || {};
+	terserOptions = {
+		ecma: chooseTerserEcmaVersion( supportedBrowsers ),
+		ie8: false,
+		safari10: supportedBrowsers.some(
+			browser => browser.includes( 'safari 10' ) || browser.includes( 'ios_saf 10' )
+		),
+		...terserOptions,
+	};
+
+	return [ new TerserPlugin( { ...options, terserOptions } ) ];
+};
```

Comment:
Heads up @Automattic/gutenpack :-)

Changes in `packages/calypso-build/webpack/minify.js`

Reply:
Good call mentioning this, @simison. 👍 

To explain what's going on: the minify module will now try to configure Terser automatically based on the available `browserslist` config. Any explicitly provided `options` and `options.terserOptions` will be preserved, however, so this shouldn't cause any changes unless you're relying on defaults and those defaults are different from your browserslist config.